include/networkit/centrality/TopHarmonicCloseness.hpp

```diff
@@ -41,8 +41,12 @@ class TopHarmonicCloseness final : public Algorithm {
      * @param k Number of nodes with highest harmonic closeness that have to be found.
      * For example, k = 10, the top 10 nodes with highest harmonic closeness will be computed.
      * @param useNBbound If true, the NBbound variation will be used, otherwise NBcut.
+     * @param nodeList Restrict closeness computation to a group of nodes from the graph.
+     * If the list is empty (default), all nodes from the graph are used for computation.
+     * Note: Actual existence of included nodes in the graph is not checked.
      */
-    explicit TopHarmonicCloseness(const Graph &G, count k = 1, bool useNBbound = false);
+    explicit TopHarmonicCloseness(const Graph &G, count k = 1, bool useNBbound = false,
+                                  const std::vector<node> &nodeList = std::vector<node>());
```

Comment:
This seems a quite specific feature and I think that adding an additional parameter to the constructor to use it is not a good idea. Could we maybe set the list via a separate function (e.g., `void restrictTopKComputationToNodes(const std::vector<node> &nodeList);`)? One could call that function before `run`.

Reply:
I also thought about whether to integrate it into the constructor or having a calling function, however came up with the opposite conclusion initially. Your argument makes sense though.